### ISSUE TITLE
[Bug]: Custom storybook theme does not work in v7.0.0-beta.31 and vite builder

### ISSUE THREAD

**Issue by karrui** [storybookjs/storybook#20704]
>Edit, updated to beta.31, no change. Themes are still not applying

### Describe the bug

using 7.0.0-beta.31 with vite builder. manager.ts does not seem to be referenced and thus the custom theme is not applied.

is anyone facing this issue?

repository is open sourced, see https://github.com/opengovsg/design-system/tree/main/react/.storybook

The built storybook (by chromatic) is also hosted on https://61ded3112e6b2c004aedabea-lnbjyofnhw.chromatic.com/ if you want to look at the output.
how it should look like: https://61ded3112e6b2c004aedabea-yifmbgvqmo.chromatic.com/

(this was before the upgrade to v7, and the storybook config can be found here: https://github.com/opengovsg/design-system/tree/release-v0.0.13-beta.5/react/.storybook)

### System

```shell
Environment Info:

  System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1 Pro
  Binaries:
    Node: 18.12.0 - ~/.nvm/versions/node/v18.12.0/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Firefox: 93.0
    Safari: 16.2
  npmPackages:
    @storybook/addon-a11y: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/addon-essentials: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/addon-links: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/addons: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/client-api: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/react: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/react-vite: ^7.0.0-beta.31 => 7.0.0-beta.31
    @storybook/theming: ^7.0.0-beta.31 => 7.0.0-beta.31
```

**Comment by Dschungelabenteuer** [comment]
I ran into this issue as well. It's worth noting that it works correctly [on Storybook's sandboxes](https://github.com/storybookjs/sandboxes). I've started investigating a bit yesterday and will take a closer look tomorrow :)

**Comment by Dschungelabenteuer** [comment]
Here is what I came up with for the moment, comparing the sandbox to my case:
* The issue occurs no matter what the extension is for the `manager` file so I'll stick to `manager.ts` as well.
* All manager entries (including the one based on our `manager.ts` we're interested in) are correctly picked and [passed in the esbuild config](https://github.com/storybookjs/storybook/blob/next/code/lib/builder-manager/src/index.ts#L45-L46) which is eventually used to bundle them.
* However, the esbuild output for our `manager.ts` file is an empty bundle only including both [`banner`](https://github.com/storybookjs/storybook/blob/next/code/lib/builder-manager/src/index.ts#L87) and [`footer`](https://github.com/storybookjs/storybook/blob/next/code/lib/builder-manager/src/index.ts#L90). You can see that [on your own published storybook as well](https://61ded3112e6b2c004aedabea-lnbjyofnhw.chromatic.com/sb-addons/.storybook/manager-bundle.mjs). It explains why our custom themes aren't used at all, but we'll still have to figure out why.
* I've compared both the sandbox and my case's esbuild configs, there is no difference. Both `manager.ts` files are specified within `entryPoints`, but mine is not actually processed (I've made a syntax error on purpose so that esbuild fails whenever it processes it).
* So I guess the cause might be around [one of these plugins](https://github.com/storybookjs/storybook/blob/next/code/lib/builder-manager/src/index.ts#L77-L83), but I'm not quite sure yet.
* When removing the [`globalExternals`](https://github.com/storybookjs/storybook/blob/next/code/lib/builder-manager/src/index.ts#L82) plugin, both Storybooks fail to start (as we could expect), BUT both `manager` files are actually processed, I guess we're getting closer.
* Comparing the output of this plugin, again there is no difference. No real surprise as it basically just write export statements based on the options it receives. I first thought the main issue was that some of the [original `definitions` keys](https://github.com/storybookjs/storybook/blob/next/code/ui/manager/src/globals/types.ts#L4) could not be resolved on our side, but it turns out as a dead end.
* Well, while digging into esbuild's verbose logs it turns out the `manager.ts` import is supposed to be "under the effect of" [this file](https://github.com/storybookjs/storybook/blob/next/code/lib/builder-manager/templates/addon.tsconfig.json), which cannot work because both our configurations prevent that:
  *  [You disable sideEffects](https://github.com/opengovsg/design-system/blob/main/react/package.json#L7)
  * I do filter them with that way: `"sideEffects": ["*.css", "*.scss"],`

**TL;DR:** Try to remove that `sideEffect` (or set it to `true`) and please LMK if it works (you might need to clear your browser cache). It worked perfectly on my side!

Summoning @ndelangen which I think worked on that part? Is it expected, or does it deserve any note/warning in [Storybook's theming guide](https://storybook.js.org/docs/7.0/react/configure/theming)?

**Comment by karrui** [comment]
> Try to remove that `sideEffect` (or set it to `true`) and please LMK if it works, @karrui (you might need to clear your browser cache). It worked perfectly on my side!

Confirming removing `sideEffects` from `package.json` does work to resolve the custom theme, wow. Great investigating.

**Comment by ndelangen** [comment]
@jonniebigodes could you invite @Dschungelabenteuer to order a swag box? 🙏

**Comment by Dschungelabenteuer** [comment]
> @jonniebigodes could you invite @Dschungelabenteuer to order a swag box? 🙏

Thank you so much, @ndelangen, but [I already got one](https://twitter.com/yungezeit/status/1529198082734317568) 😄

**Comment by filipw01** [comment]
@ndelangen Should this be marked as closed? I'm trying out the beta myself and while the workaround works, we don't want to remove `"sideEffects": false` because we want the bundlers to do tree shaking on our package. Shouldn't this be addressed in storybook's code?

**Comment by jonniebigodes** [comment]
@Dschungelabenteuer are you ok with getting a pr up to address this issue in the theming docs by adding a small note to it so that we don't lose track of it? And @ndelangen, yep, @Dschungelabenteuer got his much-deserved swag box a while ago, as he's been more than fantastic with his contributions, ideas and overall Discord, to which I'm really grateful for it 🙏 !

**Comment by Dschungelabenteuer** [comment]
Sure, @jonniebigodes! I'll try to get this done this week and let you know. Also, thank you very much for the kind words ❤️

**Comment by judithrocketmakers** [comment]
I don't know if this counts as a terrible workaround, but as we also need tree-shaking I tried this and it worked... We don't have another file called `manager.ts` in our codebase.

```
  "sideEffects": [
    "*.scss",
    "manager.ts"
  ],
```

**Comment by filipw01** [comment]
Seems it was fixed in #21335 🥳
